src/plugins/es_ui_shared/public/request/use_request.ts

```diff
@@ -65,49 +65,53 @@ export const useRequest = <D = any, E = Error>(
     /* eslint-disable-next-line react-hooks/exhaustive-deps */
   }, [path, method, queryStringified, bodyStringified]);
 
-  const resendRequest = useCallback(async () => {
-    // If we're on an interval, this allows us to reset it if the user has manually requested the
-    // data, to avoid doubled-up requests.
-    clearPollInterval();
-
-    const requestId = ++requestCountRef.current;
-
-    // We don't clear error or data, so it's up to the consumer to decide whether to display the
-    // "old" error/data or loading state when a new request is in-flight.
-    setIsLoading(true);
-
-    const response = await sendRequest<D, E>(httpClient, requestBody);
-    const { data: serializedResponseData, error: responseError } = response;
-
-    const isOutdatedRequest = requestId !== requestCountRef.current;
-    const isUnmounted = isMounted.current === false;
-
-    // Ignore outdated or irrelevant data.
-    if (isOutdatedRequest || isUnmounted) {
-      return;
-    }
-
-    // Surface to consumers that at least one request has resolved.
-    isInitialRequestRef.current = false;
+  const resendRequest = useCallback(
+    async (asSystemRequest?: boolean) => {
```

Comment:
I think it would be better to extract the parameter from `UseRequestConfig` above in the `useRequest` call and use it here. As we are not passing it on L120 and we don't want the consumer to change the request configuration when calling `resendRequest`.

Reply:
Sorry for the terrible diff here, all I did was 1. add an `asSystemRequest` parameter to the anonymous callback function, and 2. construct the request payload using this new parameter.

Actually I don't think we want to expose this in `UseRequestConfig` --  I see it as an implementation detail. A consumer calls the `useRequest` function, which schedules itself to poll in the background. Consumers should not be able to determine whether this is treated as a system request. It's background polling, so it must be treated as such.

Reply:
OK I understand. But then when we provide that handler to the consumer should we protect it and set to `false` on L145, or is it overkill?

```js
const resendRequestForConsumer = useCallback(() => {
  return resendRequest(false);
}, [resendRequest]);

return {
    isInitialRequest: isInitialRequestRef.current,
    isLoading,
    error,
    data,
    resendRequest: resendRequestForConsumer, // Gives the user the ability to manually request data
  };
```

Reply:
Oh, _now_ I understand now what you were trying to say 😄 

It's not something we _need_ to protect. As in, there's no danger of other unintended consequences if a consumer happened to use this parameter.

However, we don't have any consumers that _require_ the ability to use this when manually requesting data. With that in mind, I think we should apply your suggestion until we have a reason to expose it.